File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        result = 0
        nums_set = set(nums)

        for num in nums:
            if num - 1 in nums_set:
                continue
            
            length = 1
            k = num

            while k + 1 in nums_set:
                length += 1 
                k += 1
            
            result = max(result, length)

        return result

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> nums_set;

        for (int num : nums) {
            if (find(nums_set.begin(), nums_set.end(), num) != nums_set.end()) { return true; }
            nums_set.insert(num);
        }

        return false;
    }
};
